test/iaas.GcpClient.spec.js

```diff
@@ -0,0 +1,538 @@
+        return Promise.delay(CONNECTION_WAIT_SIMULATED_DELAY)
+          .then(() => {
+            if (streamEventHandlers.end && _.isFunction(streamEventHandlers.end)) {
+              return streamEventHandlers.end.call(streamEventHandlers.end);
+            } else {
+              throw new Error('Event Handlers not registered in delete snapshot');
+            }
+          })
+          .then(() => {
+            return downloadJsonPromise
+              .then(response => expect(response).to.eql(JSON.parse(jsonContent)))
+              .catch(() => {
+                throw new Error('expected download to be successful');
+              });
+          });
+      });
+
+      it('file/blob download should fail', function () {
+        const downloadErrMsg = 'Download failed';
+        var downloadJsonPromise = client.downloadJson(validBlobName);
+        return Promise.delay(CONNECTION_WAIT_SIMULATED_DELAY)
+          .then(() => {
+            if (streamEventHandlers.error && _.isFunction(streamEventHandlers.error)) {
+              return streamEventHandlers.error.call(streamEventHandlers.error, downloadErrMsg);
+            } else {
+              throw new Error('Event Handlers not registered in delete snapshot');
+            }
+          })
+          .then(() => {
+            return downloadJsonPromise
+              .then(() => {
+                throw new Error('expected download to fail');
+              })
+              .catch(err => expect(err).to.eql(downloadErrMsg));
+          });
+      });
+
+      it('file/blob download should fail with NotFound error', function () {
+        var downloadJsonPromise = client.downloadJson(validBlobName);
+        return Promise.delay(CONNECTION_WAIT_SIMULATED_DELAY)
+          .then(() => {
+            if (streamEventHandlers.error && _.isFunction(streamEventHandlers.error)) {
+              return streamEventHandlers.error.call(streamEventHandlers.error, fileNotFoundResponse);
+            } else {
+              throw new Error('Event Handlers not registered in delete snapshot');
+            }
+          })
+          .then(() => {
+            return downloadJsonPromise
+              .then(() => {
+                throw new Error('expected download to fail');
+              })
+              .catch(err => expect(err).to.be.instanceof(NotFound));
+          });
+      });
+    });
+
+    describe('#ComputeOperations', function () {
+      before(function () {
+        sandbox.stub(GcpCompute.prototype, 'snapshot').returns(snapshotStub);
+      });
+      after(function () {
+        sandbox.restore();
+      });
+
+      it('snapshot should be deleted successfully', function () {
+        var deleteSnapshotPromise = client.deleteSnapshot('snapshot-1');
+        return Promise.delay(CONNECTION_WAIT_SIMULATED_DELAY)
+          .then(() => {
+            if (deleteSnapshotEventHandlers.running && _.isFunction(deleteSnapshotEventHandlers.running)) {
```

Comment:
This handler might not be required as quoted above.

Reply:
Removed.